app/src/main/java/com/asfoundation/wallet/promo_code/use_cases/ObserveCurrentPromoCodeUseCase.kt

```diff
@@ -6,9 +6,19 @@ import io.reactivex.Observable
 import javax.inject.Inject
 
 class ObserveCurrentPromoCodeUseCase @Inject constructor(
-    private val promoCodeRepository: PromoCodeRepository) {
+  private val promoCodeRepository: PromoCodeRepository
+) {
 
   operator fun invoke(): Observable<PromoCode> {
     return promoCodeRepository.observeCurrentPromoCode()
+      .flatMap { promoCode ->
+        if (promoCode.expired == false) {
+          promoCode.code?.let {
+            return@flatMap promoCodeRepository.setPromoCode(it)
```

Comment:
This call will trigger the DAO to save and updated from BE promo code (according to the code in the `PromoCodeRepository`).
So this will trigger the DAO to return a new promo.
And it will hit this `flatMap` again.
It looks like for a valid promo code this will result in an infinite loop of queries to the MS and BE.
Did you test it? 🤔